cat-loving-app/src/cats/CatDetail.tsx

```diff
@@ -0,0 +1,177 @@
+    justify-content: center; // Center vertically
+    position: relative;
+    justify-content: center;
+    padding: 20px;
+    margin: 0 auto;
+    background-color: rgba(255, 255, 255);
+    color: black;
+    border-radius: 10px;
+    width: 80%;
+    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+    ${props => props.$withModal && `
+        width: 50%;
+        top: 25vh;
+    `}
+`;
+
+const Image = styled.img`
+    position: relative;
+    max-width: 100%;
+`;
+
+const DescriptionStyled = styled.p`
+    font-size: small;
+`;
+
+const Url = styled.a`
+    color: blue;
+    font-size: small;
+`;
+
+const BreedPresentationText = styled.p`
+    font-size: small;
+`;
+
+const InformationText = styled.p`
+    font-size: small;
+`;
+
+const BreedDetailUrl: React.FC<{ breed: Breed, catId: string }> = ({ breed, catId }) => {
+    return (
+        <Url href={`breedDetail/${breed.id}/${catId}`}>
+            {breed.name}
+        </Url>
+    );
+}
+
+const CatDetail: React.FC<CatDetailProps> = ({ catId: id = '', imageUrl: url = '', withModal = true, favouriteIcon }) => {
+    let {catId} = useParams();
+    if(catId === undefined && id !== '' ) {
+        catId = id;
+    }
+    
+    const [description, setDescription] = useState<string>('');
+    const [imageUrl, setImageUrl] = useState<string>(url);
+    const [showFullText, setShowFullText] = useState<boolean>(id === '');
+    const [breeds, setBreeds] = useState<Breed[]>([]);
+    const [showErrorMessage, setShowErrorMessage] = useState<boolean>(false);
+    const [showSucessMessage, setShowSucessMessage] = useState<boolean>(false);
+
+    useEffect(() => {
+        const fetchCatDescription = async () => {
+                const data = await getCat(catId || '');
+                if(!data) {
+                    return;
+                }
+                setImageUrl(data.url);
+                if(data.breeds.length > 0) { 
+                    setDescription(data.breeds[0].description);
+                    setBreeds(data.breeds);
+                }else{
+                    setDescription('No description available');
+                }
+        };
+
+        fetchCatDescription();
+    }, [catId]);
+
+    const handleAddToFavorites = useCallback(async () => {
+        const successFullySaved = await saveToFavorites(catId || '');
+        if(successFullySaved) {
+            setShowSucessMessage(true);
+            setShowErrorMessage(false);
+        }else{
+            setShowErrorMessage(true);
+            setShowSucessMessage(false);
+        }
+    }, [catId]);
+
+    const renderText = () => {
+        if (description.length <= 150) {
+            return <DescriptionStyled>{description}</DescriptionStyled>;
+        } else {
+            if (showFullText) {
+                return (
+                    <>
+                        <DescriptionStyled>{description}</DescriptionStyled>
+                        { id!== '' && 
+                            <button onClick={(evt) => {
```

Comment:
Why do you have to use `event.stopPropagation`? The CatDetailContainer has an action-less onClick event attached.

Reply:
This was interfering with the modal. CatDetail can be rendered either inside a modal or just as a single page. In the case of a modal the event propagating up to the modal level and closing it as a result

Reply:
Hmm... Maybe that's because you have that empty event listener attached. Otherwise, I can't see how the modal can be closed by clicking on part of its content.